src/jbi/whiteboard_actions/default.py

```diff
@@ -5,36 +5,141 @@
+            status_code=201,
+        )
+
+    def event_target_bug(  # pylint: disable=too-many-locals
+        self, payload: BugzillaWebhookRequest, linked_issue_key, bug_obj
+    ):
+        """Create and link jira issue with bug, or update; rollback if multiple events fire"""
+        if linked_issue_key:
+            # update
+            fields, comments = payload.map_as_tuple_of_field_dict_and_comments()
+            jira_update_response = self.jira_client.update_issue_field(
+                key=linked_issue_key, fields=fields
+            )
+            # comment
+            jira_comment_responses = []
+            for comment in comments:
+                jira_comment_responses.append(
+                    self.jira_client.issue_add_comment(
+                        issue_key=linked_issue_key, comment=comment
+                    )
+                )
+            return JSONResponse(
+                content={
+                    "status": "update",
+                    "jira_update_response": jira_update_response,
+                    "jira_comment_responses": jira_comment_responses,
+                },
+                status_code=201,
+            )
+        # else: create jira issue
+        fields = bug_obj.get_jira_issue_dict()
+        fields["project"] = {"key": self.jira_project_key}
+
+        response = self.jira_client.create_issue(fields=fields)
+
+        if isinstance(response, list):
+            response = response[0]
+
+        if isinstance(response, dict):
+            if any(
+                element in ["errors", "errorMessages"] and response[element]
+                for element in response.keys()
+            ):
+                # Failure to create: err keys exist with value
+                raise ActionError(f"response contains error: {response}")
+
+        jira_key_in_response = response.get("key")
+        if not payload.bug:
+            raise ActionError("payload does not contain bug data")
+        current_bug_info = self.bugzilla_client.getbug(payload.bug.id)
+        bug_obj = BugzillaBug.parse_obj(current_bug_info.__dict__)
+        jira_key_in_bugzilla = bug_obj.extract_from_see_also()
+        _duplicate_creation_event = (
+            jira_key_in_bugzilla is not None
+            and jira_key_in_response != jira_key_in_bugzilla
+        )
+        if not _duplicate_creation_event:
+            jira_url = f"{self.settings.jira_base_url}/browse/{jira_key_in_response}"
+            update = self.bugzilla_client.build_update(see_also_add=jira_url)
+            bugzilla_response = self.bugzilla_client.update_bugs([bug_obj.id], update)
 
-    def __call__(self, payload):
-        """
-        Called from BZ webhook
-        Call Jira SDK with project key etc.
-        """
-        print(payload)
-        print(self.parameters)
+            bugzilla_url = (
+                f"{self.settings.bugzilla_base_url}show_bug.cgi?id={bug_obj.id}"
+            )
```

Comment:
`BUGZILLA_URL = f"{self.settings.bugzilla_base_url}show_bug.cgi?id=%s"`

...

`bugzilla_url = BUGZILLA_URL.format(bug_obj.id)`

Reply:
Moved to settings object